### DISCUSSION TITLE
how to set that Opening a tab/pane in the same directory in Windows Terminal(oh-my-posh)

### DISCUSSION THREAD

**Discussion by sixwaaaay** [JanDeDobbeleer/oh-my-posh#1532] [Troubleshoot] [answered]
I followed the instruction on [how to set that Opening a tab/pane in the same directory in Windows Terminal](https://docs.microsoft.com/en-us/windows/terminal/tutorials/new-tab-same-directory)
I add this to my $profile.
```
function prompt
{
  $loc = Get-Location

  $prompt = & $GitPromptScriptBlock

  $prompt += "$([char]27)]9;12$([char]7)"
  if ($loc.Provider.Name -eq "FileSystem")
  {
    $prompt += "$([char]27)]9;9;`"$($loc.Path)`"$([char]7)"
  }

  $prompt
}
```
but when I restart the window terminal , oh my posh does't work .
how can I make it work

**Comment by JanDeDobbeleer** [comment]
You can't use that snippet as it overrides the current prompt (and thus removes oh-my-posh). You can however set [osc99](https://ohmyposh.dev/docs/config-overview#general-settings) to true when using Windows Terminal.

**Reply by eglavin** [reply]
@JanDeDobbeleer Just wondering how I set this option? Is there any way to extend a theme without creating a copy like whats [recommended in the docs?](https://ohmyposh.dev/docs/installation/customize#adjust-a-theme)

**Reply by JanDeDobbeleer** [reply]
@eglavin no, you need a local copy to adjust.

**Reply by eglavin** [reply]
Perfect thanks for the quick reply!

**Comment by pranavpa8788** [comment]
@JanDeDobbeleer I've added "pwd": "osc99" to the theme's json file which I'm using right now and it still doesn't open the new tab in the same directory, also tried "osc99": true but no luck

**Reply by JanDeDobbeleer** [reply]
Works on my end, you need to duplicate the tab or split the tab for this to work though. It does not work when you add a new tab.

**Reply by pranavpa8788** [reply]
I have my newtab keybinding set to duplicate the tab. But even when I do it manually (duplicate tab/split tab) it doesn't work. I've tried 
`"pwd": "osc99"`
and 
`"osc99": true"`
and all kinds of combination of the above two (including uppercase and lowercase) and none of them worked. 
oh my posh version: 11.3.1
windows 11 (22H2 22621.521)
windows terminal version: 1.15.2713.0
powershell version: 7.2.6
oh-my-posh theme: capr4n

Below is the oh-my-posh init line in my powershell profile
$env:Path += ';C:\Users\my_user\AppData\Local\Programs\oh-my-posh\bin\'

oh-my-posh init pwsh --config 'oh_my_posh_capr4n.omp.json' | Invoke-Expression

And below is the oh_my_posh_capr4n.omp.json file
{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "blocks": [
    {
      "alignment": "right",
      "segments": [
        {
          "background": "#003543",
          "foreground": "#ffffff",
          "style": "plain",
          "template": " \uf64f {{ .CurrentDate | date .Format }} ",
          "type": "time"
        },
        {
          "background": "#83769c",
          "foreground": "#ffffff",
          "properties": {
            "always_enabled": true
          },
          "style": "plain",
          "template": " \ufa1e{{ .FormattedMs }} ",
          "type": "executiontime"
        }
      ],
      "type": "rprompt"
    },
    {
      "alignment": "left",
      "segments": [
        {
          "background": "#61AFEF",
          "foreground": "#ffffff",
          "properties": {
            "display_host": false
          },
          "style": "diamond",
          "template": "{{if .Root}} \uf0e7 {{.UserName}} {{else}} {{.UserName}} {{end}}",
          "trailing_diamond": "\ue0b0",
          "type": "session"
        },
        {
          "background": "#C678DD",
          "foreground": "#ffffff",
          "powerline_symbol": "\ue0b0",
          "properties": {
            "folder_icon": "\uf115",
            "folder_separator_icon": " \ue0b1 ",
            "max_depth": 2,
            "style": "agnoster_short"
          },
          "style": "powerline",
          "template": " {{ .Path }} ",
          "type": "path"
        },
        {
          "background": "#95ffa4",
          "foreground": "#193549",
          "powerline_symbol": "\ue0b0",
          "style": "powerline",
          "template": " {{ .HEAD }} ",
          "type": "git"
        }
      ],
      "type": "prompt"
    }
  ],
  "console_title_template": "📁{{.Folder}}",
  "final_space": true,
  "version": 2,
  "pwd": "osc99"
}

**Reply by JanDeDobbeleer** [reply]
@pranavpa8788 don't know what happens here. The setting is correct and toggling that on my end makes a difference in working and non working directory persistence. Oh-my-posh will always add that context, so maybe there's a Windows Terminal setting that toggles this.

**Comment by MartyIX** [comment, marked as answer]
I'll summarize it for me to do it faster next time.

1. Make sure your `C:\Users\<user>\Documents\PowerShell\Microsoft.PowerShell_profile.ps1` profile contains a line like `oh-my-posh init pwsh --config "$env:POSH_THEMES_PATH/unicorn.omp.json" | Invoke-Expression`
2. Use `code $env:POSH_THEMES_PATH/unicorn.omp.json` to open the file VS code (or any other editor)
3. Modify the file as follows:
    ```diff
    {
      "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
    + "pwd": "osc99",
      "blocks": [
        {
          "alignment": "left",
        ...
    ```

See also
* https://learn.microsoft.com/en-us/windows/terminal/tutorials/new-tab-same-directory
* https://ohmyposh.dev/docs/configuration/overview#general-settings - for `pwd` settings (`osc99`)

**Reply by jloa** [reply]
doesnt work for me

**Reply by o3wiz** [reply]
You're amazing!!

**Comment by macsla** [comment]
Additional context:
This doesn't seem to work when using split pane, split: down (alt+shift+minus) and split pane, split: right (alt+shift+plus) shortcuts. 
However, duplicate pane (alt+shift+d) and duplicate tab (ctrl+shift+d) work just fine.

**Reply by aleksmark98** [reply]
You need to add a custom action for splitting the pane and set the "splitMode" to "duplicate".
I've added the following actions to overwrite default behavior and make split panes launch in the same directory as the current pane.
``` 
      {
            "command": 
            {
                "action": "splitPane",
                "split": "horizontal",
                "splitMode": "duplicate"
            },
            "keys": "alt+shift+-"
        },
        {
            "command": 
            {
                "action": "splitPane",
                "split": "vertical",
                "splitMode": "duplicate"
            },
            "keys": "alt+shift+plus"
        },
```

**Comment by shanselman** [comment]
Script to automate this here https://github.com/shanselman/splitpanefix
